### PR TITLE
Arc Primitives

### LINKED ISSUE #227: How do I draw an arc?
This is either missing from the documentation or I am unable to find how to do this.  If I want to draw an arc (that represents part of the circumference of a circle), how would I do that with Compose.jl?  Thanks!

### PR DESCRIPTION
**Pull request by Mattriks**
### This PR:
- [x] Adds `Compose.arc/slice` for drawing arcs and pie slices (closes #227)  
* The method is `arc(x, y, r, angle1, angle2, slice::Bool)`.   
* `slice` is `arc` with `slice=true`.   
* Arcs are consistently drawn clockwise from `angle1` to `angle2`.   

- [x] Works for Cairo, SVG & PGF backends
- [x] I've also removed some out-of-date Path Primitive code (see Example 2 below for more info)  
- [x] Basic documentation
 - [x]  Test implemented

### Examples:
```julia
# Example 1

colv = ["red","orange","green","blue", "purple"]
a = range(-π/4, stop=7π/4, length=6)+ 0.2*randn(6)
a[6] = a[1]

# arc/slice can toke scalars or vectors
img1 = compose(context(),
    (context(), slice([0.5], [0.5], [0.3], a[1:5], a[2:6]), fill(colv)) )
img2 = compose(context(),
    (context(), slice([0.5], [0.5], [0.32], a[1:5], a[2:6]), stroke("white"), fill(colv), linewidth(1.4mm)) )
img3 = compose(context(),
    (context(), slice([0.5], [0.5], [0.3], a[1:5], a[2:6]), stroke(colv), fill("transparent"), linewidth(1.4mm)) )
# Default arg slice = false:
img4 = compose(context(),
    (context(), arc([0.5], [0.5], [0.3], a[1:5], a[2:6]), stroke(colv), fill("transparent"), linewidth(2mm)) )

draw(PNG(8inch,2inch), hstack(img1, img2, img3, img4))
```
arcex1

In Example 2, I reproduce the example from the gist in #116. That gist uses old Path Primitive code, which is undocumented, hasn't been maintained, and is not user-friendly (the gist has ~100 lines of user code). 
Here is my user-friendly code:
```julia
# Example 2

a = range(-0.5π, stop=1.5π, length=13)
colv = repeat(["white","black"],outer=6)

 img = compose(context(),
        (context(), slice([0.5], [0.5], [0.4], a[1:12], a[2:13]), fill(colv)),
        (context(order=2), arc([0.5], [0.5], [0.2], [0, π], [π,0]), 
            fill(["black","white"]), stroke(["white","black"]), linewidth(6pt))
    ) 
draw(PNG(3inch,3inch), img)
```
arcex2

### CODE REVIEW

**Comment by tlnagy** [comment]
I gave this a quick glance over and it looks great! The biggest thing that I would add are better docstrings for the user facing functions of slice and arc. They should describe the parameters and provide one or two examples of usage.

**Comment by Mattriks** [comment]
I've refactored the Compose docs Gadfly-style, and put examples of `arc` and `slice` in the new Compose gallery. The Compose docs need a bit of work, so I'll leave adding examples of the other forms to a future PR. For this current PR I still have 2 other items on my task list.

File: src/form.jl [hunk 1]
```diff
@@ -575,408 +575,83 @@ boundingbox(form::BitmapPrimitive, linewidth::Measure, font::AbstractString, fon
...
+    slice(x, y, r, θ1, θ2)
 
-resolve(box::AbsoluteBox, units::UnitBox, t::Transform, p::QuadCurveRelPathOp) =
-        QuadCurveRelPathOp(
-            (resolve(box, units, t, p.ctrl1[1]),
-             resolve(box, units, t, p.ctrl1[2])),
-            (resolve(box, units, t, p.to[1]),
-             resolve(box, units, t, p.to[2])))
+Define a pie slice with its center at (`x`,`y`), radius of `r`, between `θ1` and `θ2`  
+"""
```
[the remark this answers is not in the document]
Comment (Mattriks) [reply src/form.jl]: Examples are in the new Compose gallery (see my comment below), rather than in docstrings: same as in Gadfly.

**Comment by Mattriks** [comment]
Done!  `Gadfly.Geom.pie` anyone?

File: src/abandoned.jl [hunk 2]
```diff
@@ -0,0 +1,816 @@
+
```
Comment (tlnagy) [inline src/abandoned.jl]: Is this all stuff that isn't needed any more? Why are we keeping it?

Comment (Mattriks) [reply src/abandoned.jl]: At the moment, I can't think of a reason why you would want Path Primitives, in addition to the Compose forms (line, curve etc). If I don't think of a reason for wanting it, I'll remove the abandoned code in a separate PR.

Comment (tlnagy) [reply src/abandoned.jl]: Yeah, that'll give us time to think of a good way to handle it and we can merge this when you do

Comment (tlnagy) [reply src/abandoned.jl]: Sorry i wasn't clear. I meant that those as two separate things, i.e. handling the abandoned code in separate PR and merging this PR now.

**Comment by bjarthur** [comment]
https://github.com/GiovineItalia/Compose.jl/blob/master/src/cairo_backends.jl#L469 is the only call to libcairo in all of Compose.jl.  could the containing function also be be moved to src/abandoned.jl?

**Comment by Mattriks** [comment]
I agree, `current_point` can be moved to abandoned.jl, and abandoned.jl can be removed. Great to see the Compose code being tidied up.

**Comment by bjarthur** [comment]
`slice` might get confused with `sortslices` and `mapslices`.  might `wedge` be a better name?

**Comment by Mattriks** [comment]
Wedge has different [geometric definition](https://en.wikipedia.org/wiki/Wedge_(geometry)). But I would agree to [sector](https://en.wikipedia.org/wiki/Circular_sector).

**Comment by tlnagy** [comment]
Do we want to make this change before we tag a new release?

**Comment by Mattriks** [comment]
I see now that what is needed a new Compose form e.g. `Compose.beziergon` (or `bezigon`). That should be possible (at least in svg).

**Pull request merged**